packages/cli/src/modules/mcp/tools/search-workflows.tool.ts

```diff
@@ -0,0 +1,109 @@
+import { type User, type WorkflowEntity } from '@n8n/db';
+import type { INode } from 'n8n-workflow';
+import z from 'zod';
+
+import type {
+	ToolDefinition,
+	SearchWorkflowsParams,
+	SearchWorkflowsResult,
+	SearchWorkflowsItem,
+} from '../mcp.types';
+
+import type { ListQuery } from '@/requests';
+import type { WorkflowService } from '@/workflows/workflow.service';
+
+const MAX_RESULTS = 200;
+
+const inputSchema = {
+	limit: z
+		.number()
+		.int()
+		.positive()
+		.max(MAX_RESULTS)
+		.optional()
+		.describe(`Limit the number of results (max ${MAX_RESULTS})`),
+	active: z.boolean().optional().describe('Filter by active status'),
+	name: z.string().optional().describe('Filter by name'),
+	projectId: z.string().optional(),
+} satisfies z.ZodRawShape;
+
+export const createSearchWorkflowsTool = (
+	user: User,
+	workflowService: WorkflowService,
+): ToolDefinition<typeof inputSchema> => {
+	return {
+		name: 'search_workflows',
+		config: {
+			description:
+				'Search for workflows with optional filters. Returns a preview of each workflow.',
+			inputSchema,
+		},
+		handler: async ({ limit = MAX_RESULTS, active, name, projectId }) => {
+			const payload: SearchWorkflowsResult = await searchWorkflows(user, workflowService, {
+				limit,
+				active,
+				name,
+				projectId,
+			});
+
+			return {
+				content: [
+					{
+						type: 'text',
+						text: JSON.stringify(payload),
+					},
```

Comment:
Not familiar with MCP, but does it make sense to return everything in a single text or should we return multiple items in the outer array?

Reply:
There is also a [structured output support](https://modelcontextprotocol.io/specification/2025-06-18/server/tools#structured-content) in MCP so makes sense to add it here. But looks like we cannot get rid of text output:

> For backwards compatibility, a tool that returns structured content SHOULD also return the serialized JSON in a TextContent block.